### ISSUE TITLE
Feature request: single-click, no-dialog, Add download option

### ISSUE THREAD

**Issue by jwmh** [TachibanaGeneralLaboratories/download-navi#62]
Feature request.

# summary
Add option for single-tap, zero-dialogue, immediate-Add downloads

... especially, when links are Shared into Navi, from other apps (e.g. from Chrome or Firefox).

... which **immediately** Adds the link as a background download.
Zero interaction.

## Use case scenario
 (where all of the following are all true, all at the same time).

- I'm in some app (e.g. PDF viewer, or Chrome or Firefox, or Gmail, or whatever).

- I'm reading content -- long articles or documents -- with other links to other pages.

- I may be online, or offline, or toggling between both, in order to save battery.
(In most cases, I'm mostly offline, no internet connection.)

- I am adding new downloads / save for offline URLs on average *one every few seconds* (multiple times per minute!).

-- I do this by hitting the "Share..." button (inside whatever app I'm in), then tapping the "Download Navi" button.

## Navi's current actual behavior
It opens the full dialogue box, with fields to type into (the url is prefilled).
I have to select options.
I have to tap either Add or Connect. (However I have the Setting set so that "Add" converts to Connect, if I am online.)

## feature request 
Provide a single-tap // zero-dialogue choice from the "Share..." menu of other apps.
Behavior:
I click "Share...", choose "DL Navi Now" (or something similar), and the link is IMMEDIATELY added in the background with no further action.  I immediately continue reading my article or document, adding more background downloads as I go.

## Means of implementation
... Rather than making this the default, may I instead suggest doing this:
A.) either via add a user-toggle (checkbox or radio-button) option in settings...
 and/or 
B.) add a second Android "Share..." activity for the app ... which then allows users to choose from the Android "Share..." menu, either:
(i.) the dialogue popup (current behavior), 
- or -
(ii.) a different second Activity, the "auto-Add-to-download-queue" activity.

Sorry if this is really unclear / a big mess.
Does this all make sense
?

**Comment by proninyaroslav** [comment]
Makes sense after implementation https://github.com/TachibanaGeneralLaboratories/download-navi/issues/32, https://github.com/TachibanaGeneralLaboratories/download-navi/issues/58
